core/arch/arm/kernel/thread_spmc.c

```diff
@@ -1167,3 +1169,19 @@ uint32_t thread_rpc_cmd(uint32_t cmd, size_t num_params,
 
 	return get_rpc_arg_res(arg, num_params, params);
 }
+
+struct mobj *thread_rpc_alloc_global_payload(size_t size __unused)
+{
+	return NULL;
+}
+
+void thread_rpc_free_global_payload(struct mobj *mobj __unused)
+{
+	/*
+	 * "can't happen" since thread_rpc_alloc_global_payload() always
+	 * returns NULL.
+	 */
+	volatile bool cant_happen __maybe_unused = true;
+
+	assert(!cant_happen);
```

Comment:
This is a bit convoluted, why not `assert(false);`? Compiler warninigs perhaps?

Reply:
Yeah, compiler warning is the problem. If we had a `__maybe_noreturn` this wouldn't be needed.